manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -188,6 +188,9 @@ func (tr *TaskReaper) tick() {
 		for dirty := range tr.dirty {
 			service := store.GetService(tx, dirty.ServiceID)
 			if service == nil {
+				// If the service can't be found, assume that it was deleted
+				// and remove the task from the dirty list.
+				delete(tr.dirty, dirty)
```

Comment:
Should this be moved to `cleanup` instead of just being deleted from `dirty`, otherwise they'll never be removed from the store until the next leader election?

Reply:
Er sorry `deleteTasks`

Reply:
Negative. the `dirty` map is orchestrator.SlotTuple objects, which represent a service and a node or slot, not an individual task. The task itself should be in the `cleanup` list already.

Reply:
That said, the comment ought to be reworded to say `remove the slot` instead of `remove the task`

Reply:
Ah ok, thanks!